### PR TITLE
Fix bounds check instrumentation to handle 16-bit values

### PR DESCRIPTION
**Pull request by greg-lunarg**
Fixes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/2234

### CODE REVIEW

**Comment by greg-lunarg** [comment]
Not sure why changing 4 instrumentation files would break spirv-val. I believe this branch has the most recent master commits as of noon our time.

**Comment by greg-lunarg** [comment]
test\val\Debug\test_val_abcde.exe builds on my machine with VS2017

File: source/opt/inst_bindless_check_pass.cpp [hunk 1]
```diff
@@ -434,18 +436,27 @@ void InstBindlessCheckPass::GenCheckCode(
...
+    // modes will use same debug stream write function
+    uint32_t u_length_id = GenUintCastCode(length_id, &builder);
     GenDebugStreamWrite(
         uid2offset_[ref->ref_inst->unique_id()], stage_idx,
-        {error_id, u_index_id, length_id, builder.GetUintConstantId(0)},
+        {error_id, u_index_id, u_length_id, builder.GetUintConstantId(0)},
         &builder);
-  else
+  } else {
+    // Uninitialized Descriptor - Normal erro return
```
Comment (dnovillo) [inline source/opt/inst_bindless_check_pass.cpp:455]: nit: "Normal erro" -> "Normal error"

File: source/opt/instrument_pass.h [hunk 2]
```diff
@@ -213,8 +213,12 @@ class InstrumentPass : public Pass {
   uint32_t GenDebugDirectRead(const std::vector<uint32_t>& offset_ids,
                               InstructionBuilder* builder);
 
-  // Generate code to cast |value_id| to unsigned, if needed. Return
-  // an id to the unsigned equivalent.
+  // Generate code to convert integer |value_id| to 32bit, if needed. Return
+  // an id to the 32bit equivalent.
+  uint32_t Gen32BitCvtCode(uint32_t value_id, InstructionBuilder* builder);
+
+  // Generate code to cast integer|value_id| to 32bit unsigned, if needed.
```
Comment (dnovillo) [inline source/opt/instrument_pass.h:220]: nit: "integer|value_id|" -> "integer |value_id|"

**Review by dnovillo — APPROVED** [review]
Apologies that my previous comment wasn't bundled with this one.  Patch LGTM.  Thanks.

File: source/opt/inst_bindless_check_pass.cpp (hunk 1 above)
Comment (greg-lunarg) [reply source/opt/inst_bindless_check_pass.cpp]: Fixed.

File: source/opt/instrument_pass.h (hunk 2 above)
Comment (greg-lunarg) [reply source/opt/instrument_pass.h]: Fixed. Thanks for the review!

**Comment by greg-lunarg** [comment]
@dnovillo I believe all issues have been addressed. Thanks for the review!

**Pull request merged**